src/openrct2/ride/Ride.cpp

```diff
@@ -1132,13 +1132,11 @@ void ride_remove_peeps(Ride* ride)
 
             if (exitDirection == 255)
             {
-                int32_t x = peep->next_x + 16;
-                int32_t y = peep->next_y + 16;
-                int32_t z = peep->next_z * 8;
+                CoordsXYZ newLoc = { peep->NextLoc.ToTileCentre(), peep->NextLoc.z };
```

Comment:
I think that just doing

```suggestion
                CoordsXYZ newLoc = peep->NextLoc.ToTileCentre();
```

should work.

Reply:
Nah that just creates a cord xy